dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs

```diff
@@ -139,6 +140,10 @@ public sealed class FunctionCallingStepwisePlanner
                     {
                         chatHistoryForSteps.AddUserMessage($"Failed to execute function {functionResponse.FullyQualifiedName}. Try something else!");
                     }
+                    catch (Exception ex) when (!ex.IsCriticalException())
```

Comment:
Can we add a test case for this functionality?

Reply:
Working on it!